### PR TITLE
Update to use level 8

### PR DESCRIPTION
**Pull request by finwo**
Includes:
- https://github.com/levelgraph/levelgraph/pull/198
- https://github.com/levelgraph/levelgraph/pull/196

Extras:
- Adds a polyfill for levelup.createValueStream (and related methods), as that has been ditched with level 8 using it's default classic-level export

### CODE REVIEW

**Comment by vweevers** [comment]
FYI for the streams, see https://github.com/Level/read-stream

**Comment by finwo** [comment]
> FYI for the streams, see https://github.com/Level/read-stream

Awesome, I'll try to implement that tomorrow instead of the custom workaround

**Comment by finwo** [comment]
Just tried using level-read-stream, but I ran into the following error:

```
     Uncaught TypeError: db[method] is not a function
      at new LevelReadStream (node_modules/level-read-stream/index.js:17:33)
      at new ValueStream (node_modules/level-read-stream/index.js:80:5)
      at Object.getStream (lib/levelgraph.js:60:20)
      at JoinStream.transform [as _transform] (lib/joinstream.js:95:30)
      at Transform._write (node_modules/readable-stream/lib/internal/streams/transform.js:153:8)
      at writeOrBuffer (node_modules/readable-stream/lib/internal/streams/writable.js:334:12)
      at _write (node_modules/readable-stream/lib/internal/streams/writable.js:283:10)
      at Writable.end (node_modules/readable-stream/lib/internal/streams/writable.js:539:17)
      at /Users/finwo/src/finwo/levelgraph/lib/levelgraph.js:150:18
      at /Users/finwo/src/finwo/levelgraph/lib/queryplanner.js:93:7
      at NoResultsHolder.release (node_modules/fastparallel/parallel.js:139:9)
      at process.processTicksAndRejections (node:internal/process/task_queues:77:11)
```

This leads me to believe the older LevelUP interface is not supported by the level-read-stream package.

I'll certainly have to fix this some time, but for now I'd say that's out of scope for this PR, as the workaround supports both the older and newer interface.

**Comment by vweevers** [comment]
Was that when using `level-mem`? That needs to be replaced as well. I wouldn't recommend running old (e.g. `level-mem`) and new (e.g. `level@8`) side by side. See the table at https://github.com/Level/community?tab=readme-ov-file#how-do-i-upgrade-to-abstract-level

**Comment by finwo** [comment]
Correct, that was using level-mem. Using level@8 that could simply be replaced by 'new Level()'?

**Comment by vweevers** [comment]
I assume that the tests here use `level-mem` as a lightweight in-memory db, while real users would more likely use `level`. Meaning you want those databases to have the same interface.

If so then replace:

```js
const mem = require('level-mem')
const db = mem()
```

With:

```js
const { MemoryLevel } = require('memory-level')
const db = new MemoryLevel()
```

**Comment by finwo** [comment]
Gotcha, will test locally and commit if that'll work including `level-read-stream`

**Comment by finwo** [comment]
@mcollina Still in doubt on the changelog that scenaristaur added, as it's current state doesn't really make sense to me.

Would the current state of this PR be fine by you?

If so, the version bump would probably have to be a major, as it's breaking changes for people on older versions of leveldb.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# level has been upgraded
```
Comment (mcollina) [inline CHANGELOG.md:1]: please remove this file

**Review by mcollina — COMMENTED** [review]
Amazing Work!

File: CHANGELOG.md (hunk 1 above)
Comment (finwo) [reply CHANGELOG.md]: Done, will start on updating the readme.

**Comment by mcollina** [comment]
Can you update CI to support only Node v18 an v20? I'll bump a major anyway.

**Comment by finwo** [comment]
Sure, I'll make it run 18/20

**Comment by finwo** [comment]
One more issue I have with the contributor table: it seems jez0990 has removed their github profile. Should we hunt for their current online presence or just leave it as-is?

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**